src/main/java/formflow/library/pdf/DatabaseFieldPreparer.java

```diff
@@ -17,20 +17,23 @@ public class DatabaseFieldPreparer implements DefaultSubmissionFieldPreparer {
   public Map<String, SubmissionField> prepareSubmissionFields(Submission submission, PdfMap pdfMap) {
     Map<String, SubmissionField> databaseFields = new HashMap<>();
     Map<String, Object> dbFields = pdfMap.getDbFields();
-
-    dbFields.forEach((fieldName, value) -> {
-      switch (fieldName) {
-        case "submittedAt" ->
-            databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getSubmittedAt())));
-        case "submissionId" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getId().toString()));
-        case "createdAt" ->
-            databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getCreatedAt())));
-        case "updatedAt" ->
-            databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getUpdatedAt())));
-        case "flow" -> databaseFields.put(fieldName, new DatabaseField(fieldName, submission.getFlow()));
-        default -> log.error("Unable to map unknown database field: {}", fieldName);
-      }
-    });
+    
+    if (dbFields != null) {
+      dbFields.forEach((fieldName, value) -> {
+        switch (fieldName) {
+          case "submittedAt" ->
+              databaseFields.put(fieldName, new DatabaseField(fieldName, formatDateWithNoTime(submission.getSubmittedAt())));
```

Comment:
I realize that my comment has nothing to do with _this_ story. Looking around, it feels awkward that the returned `Map` duplicates the `fieldName` in the key and in the `name` property of the value.